source/products.html

```diff
@@ -1,13 +1,21 @@
 {% if categories.active != blank or artists.active != blank %}
   <div class="artist-category-nav artist-category-nav-header">
     <nav aria-label="Category navigation" class="category-nav products-nav">
-      <div class="nav-title">Products</div>
+      <div class="nav-title">
+        {% if page.full_url contains "search=" %}
+          Search Results
+        {% else %}
+          {{ page.name }}
+        {% endif %}
+      </div>
+      {% unless page.full_url contains 'search=' %}
       <ul>
         <li class="{% if page.full_url contains '/products' %}selected{% endif %}"><a href="/products">All</a></li>
         {% for category in categories.active %}
           <li class="{% if page.full_url contains category.url %}selected{% endif %}">{{ category | link_to }}</li>
         {% endfor %}
       </ul>
+      {% endunless %}

```

Comment:
i think you'll need to modify this conditional so that the artists block below is hidden as well. worth testing with a store that has artists enabled.